cnd/src/storage.rs

```diff
@@ -817,3 +1376,212 @@ impl Save<ForSwap<WhatBobLearnedFromAlice<identity::Lightning, identity::Ethereu
+        let refund_ethereum_identity = swap.data.alpha_refund_identity;
+        let secret_hash = swap.data.secret_hash;
+
+        self.db
+            .do_in_transaction(|conn| {
+                self.db.update_hbit_transient_identity(
+                    conn,
+                    local_swap_id,
+                    redeem_bitcoin_identity,
+                )?;
+                self.db.update_herc20_refund_identity(
+                    conn,
+                    local_swap_id,
+                    refund_ethereum_identity,
+                )?;
+                self.db
+                    .insert_secret_hash(conn, local_swap_id, secret_hash)?;
+
+                Ok(())
+            })
+            .await
+    }
+}
+
+#[async_trait::async_trait]
+impl Save<ForSwap<WhatAliceLearnedFromBob<identity::Bitcoin, identity::Ethereum>>> for Storage {
+    async fn save(
+        &self,
+        swap: ForSwap<WhatAliceLearnedFromBob<identity::Bitcoin, identity::Ethereum>>,
+    ) -> anyhow::Result<()> {
+        let local_swap_id = swap.local_swap_id;
+        let transient_redeem_bitcoin_identity = swap.data.alpha_redeem_identity;
+        let refund_ethereum_identity = swap.data.beta_refund_identity;
+
+        self.db
+            .do_in_transaction(|conn| {
+                self.db.update_hbit_transient_identity(
+                    conn,
+                    local_swap_id,
+                    transient_redeem_bitcoin_identity,
+                )?;
+                self.db.update_herc20_refund_identity(
+                    conn,
+                    local_swap_id,
+                    refund_ethereum_identity,
+                )?;
+
+                Ok(())
+            })
+            .await
+    }
+}
+
+#[async_trait::async_trait]
+impl Save<ForSwap<WhatBobLearnedFromAlice<identity::Bitcoin, identity::Ethereum>>> for Storage {
+    async fn save(
+        &self,
+        swap: ForSwap<WhatBobLearnedFromAlice<identity::Bitcoin, identity::Ethereum>>,
+    ) -> anyhow::Result<()> {
+        let local_swap_id = swap.local_swap_id;
+        let transient_refund_bitcoin_identity = swap.data.alpha_refund_identity;
+        let redeem_ethereum_identity = swap.data.beta_redeem_identity;
+        let secret_hash = swap.data.secret_hash;
+
+        self.db
+            .do_in_transaction(|conn| {
+                self.db.update_hbit_transient_identity(
+                    conn,
+                    local_swap_id,
+                    transient_refund_bitcoin_identity,
+                )?;
+                self.db.update_herc20_redeem_identity(
+                    conn,
+                    local_swap_id,
+                    redeem_ethereum_identity,
+                )?;
+                self.db
+                    .insert_secret_hash(conn, local_swap_id, secret_hash)?;
+
+                Ok(())
+            })
+            .await
+    }
+}
+
+fn build_hbit_params(
+    seed: RootSeed,
+    hbit: Hbit,
+    id: LocalSwapId,
```

Comment:
If you change `RootSeed` for `SwapSeed`, you can remove the `LocalSwapId` parameter.

Reply:
Unless you want to use the new error structs here as well, those take a `LocalSwapId` parameter :)